### PR TITLE
Add repermission link checker

### PR DESCRIPTION
**Pull request by mario-galic**
Given magic link

```
https://profile.thegulocal.com/repermission-token/6sMVhA5D_D3e5K5QjZ_ytByZQZO1YPnuE1Xmf27L8_rjY3zKxczVO_GKNAt7ZK6S.NL3x1JLqG1BSYp3C-cJ_bw/accept
```

and clicking the `Check repermissioning link` button

image

and entering the magic link into input text field

image

on check passing return

image

while on check failing return

image

Related PR: https://github.com/guardian/identity/pull/849

### CODE REVIEW

File: app/services/IdApi.scala [hunk 1]
```diff
@@ -41,4 +40,16 @@ class IdApi @Inject() (conf: Config, wsClient: WSClient)(implicit ec: ExecutionC
       Left(CustomError("Fatal Error", contact))
     }
   }
+
+  def checkRepermissiningLink(email: String, token: String): Future[Either[CustomError, String]] = {
+    wsClient.url(s"$baseUrl/repermission/check")
+      .addHttpHeaders("X-GU-ID-Client-Access-Token" -> s"Bearer $idApiClientAccessToken")
+      .addQueryStringParameters("email" -> email, "token" -> token)
+      .get()
```
Comment (Calum-Campbell) [inline app/services/IdApi.scala]: could remove the (), just get

File: app/controllers/RepermissionChecker.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        repermissionTokenFromUrl(formData.link)
+          .map(token => doCheck(formData.id, formData.email, token))
+          .getOrElse(Future.successful(Redirect(routes.AccessUser.getUser(formData.id)).flashing("error" -> "Incorrect format of repermissioning link")))
+    )
+  }
+
+  private def doCheck(userId: String, email: String, token: String): Future[Result] = {
+    idApi.checkRepermissiningLink(email, token).map {
+      case Right(result) =>
+        if (result == "OK")
```
Comment (jfsoul) [inline app/controllers/RepermissionChecker.scala]: just put this condition in the matcher and have a third case?

File: app/controllers/RepermissionChecker.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+      formData =>
+        repermissionTokenFromUrl(formData.link)
+          .map(token => doCheck(formData.id, formData.email, token))
+          .getOrElse(Future.successful(Redirect(routes.AccessUser.getUser(formData.id)).flashing("error" -> "Incorrect format of repermissioning link")))
+    )
+  }
+
+  private def doCheck(userId: String, email: String, token: String): Future[Result] = {
+    idApi.checkRepermissiningLink(email, token).map {
```
Comment (jfsoul) [inline app/controllers/RepermissionChecker.scala]: typo in the method name

File: app/views/editUser.scala.html [hunk 4]
```diff
@@ -57,7 +57,7 @@
         }
         @if(userForm.hasErrors) {
             <div class="alert alert-danger" role="alert">
-                <p>@Messages("editUser.invalidSubmission")</p>
+                <p>Errors: </p>
```
Comment (jfsoul) [inline app/views/editUser.scala.html]: You don't want to put this in messages?

File: app/services/IdApi.scala (hunk 1 above)
Comment (mario-galic) [reply app/services/IdApi.scala]: Convention allows for omission of () if method of 0-arity has no side-effects, however Here `get()` is doing HTTP I/O. 
https://docs.scala-lang.org/style/method-invocation.html#arity-0

**Pull request merged**